generator/internal/stub_factory_generator.cc

```diff
@@ -51,9 +51,13 @@ Status StubFactoryGenerator::GenerateHeader() {
   auto result = HeaderOpenNamespaces(NamespaceType::kInternal);
   if (!result.ok()) return result;
 
+  HeaderPrint(  // clang-format off
+    "internal::Options Resolve$service_name$Options(internal::Options options);\n\n");
+  //clang-format on
+
   HeaderPrint(  // clang-format off
     "std::shared_ptr<$stub_class_name$>\n"
-    "CreateDefault$stub_class_name$($product_namespace$::$connection_options_name$ const& options);\n\n");
+    "CreateDefault$stub_class_name$(internal::Options options);\n\n");
```

Comment:
I think we're _really_ close (any day now) close to making `Options` non-internal. When we do that all these sites will need to be renamed, along with all the other call sites outside of the generator code. We'll have to do this all in one big PR.

Do you imagine that will be easy to do w/ the generator and its artifacts? Or would it be easier to hold off on this PR for a couple days until `Options` is non-internal? Either's fine with me as long as the generator code will respond well to a large rename PR.